multi-layer-app-config/README.md

```diff
@@ -0,0 +1,193 @@
+
+#### Example
+
+With empty config list, works as it was so far:
+
+```yaml
+apiVersion: application.giantswarm.io/v1alpha1
+kind: App
+spec:
+  catalog: giantswarm
+  config:
+    configMap:
+      name: ingress-controller-values
+      namespace: m2m01
+  configs: []  # <-- new
+  name: nginx-ingress-controller-app
+  namespace: kube-system
+  userConfig:
+    configMap:
+      name: nginx-ingress-controller-app-user-values
+      namespace: m2m01
+  version: 2.7.0
+```
+
+Using the new list instead of `config` and `userConfig`, works exactly the same as the one above:
+
+```yaml
+apiVersion: application.giantswarm.io/v1alpha1
+kind: App
+spec:
+  catalog: giantswarm
+  configs:  # <-- new
+    - kind: configMap
+      name: ingress-controller-values
+      namespace: m2m01
+    - kind: configMap
+      name: nginx-ingress-controller-app-user-values
+      namespace: m2m01
+  name: nginx-ingress-controller-app
+  namespace: kube-system
+  version: 2.7.0
+```
+
+Using the new list only and referencing more than 2 objects:
+
+```yaml
+apiVersion: application.giantswarm.io/v1alpha1
+kind: App
+spec:
+  catalog: giantswarm
+  configs:  # <-- new
+    - kind: configMap
+      name: ingress-controller-values
+      namespace: m2m01
+    - kind: configMap
+      name: nginx-ingress-controller-app-user-values
+      namespace: m2m01
+    - kind: secret
+      name: nginx-ingress-controller-admin-login
+      namespace: m2m01
+    - kind: configMap
+      name: nginx-ingress-controller-admin-account
+      namespace: m2m01
+  name: nginx-ingress-controller-app
+  namespace: kube-system
+  version: 2.7.0
+```
+
+#### Pros
+
+- solves the problem everywhere, in App CR and in GitOps
+- backward compatible
+- easy to implement
+- makes `app-operator` more like `helm-controller` from Flux, which may make replacing `chart-operator` with it easier
+
+#### Cons
+
+- we're solving a problem already solved in Flux's `HelmRelease`
+- we have to implement it
+
+### Implementing ConfigMap key merging in `kustomize`
+
+Currently, `kustomize` can't merge keys within a `ConfigMap`. Still, this is a 'wanted' feature. We might just
+implement this in `kustomize`.
+
+#### Pros
+
+- solves the problem for GitOps
+- fame and glory in the community for implementing a needed feature
+- getting to know `kustomize`'s code base
+
+#### Cons
+
+- we have to implement it on an unknown code base - might be a bigger challenge
+- this solves the problem for GitOps only (well, any tool using `kustomize`)
+
+### Drop App CR and switch to HelmRelease
+
+This seems unrealistic, as we really a lot on app platform and its features. Still, bypassing App CR in GitOps
+scenario would solve the problem.
+
+#### Pros
+
+- we don't implement anything in code
+
+#### Cons
+
+- this might mean phasing out app platform operators or living in two worlds (with and without app platform)
```

Comment:
not sure if phasing out is a con or maybe even a pro, if there's really not much more to current app platform use cases than what we can see already there in Flux, why should we keep it?

Sure, we have the metadata stuff in there, but there could be ways to have that on the side of or integrated in Flux.

I think similar to phasing out vintage for CAPI a decision to phase out our operators could be a strategic pro in the end.

Reply:
cc @teemow

Reply:
We discussed this option multiple times in our team. I think pivoting makes sense but is not the right decision at this very moment. Just my 2c though

Reply:
I still think app platform has at least two unique values that still make it valid: real API representation of Helm charts and their deployments and extended meta data that we can use for better user experience. Also, dropping it would be a major effort right now, so I also think we need app platform right now.